File: ConsolePR2_3.cpp

#include <iostream>
#include <stdio.h>


#include <tuple>
using namespace std;

tuple<int, int, int> generEuclidAlg(int a, int b) {
    if (b == 0) {
        return make_tuple(a, 1, 0);
    } else {
        int d, x, y;
        tie(d, x, y) = generEuclidAlg(b, a % b);
        return make_tuple(d, y, x - (a / b) * y);
    }
}

int findMultIn(int d, int m) {
    int s, _;
    tie(_, s, _) = generEuclidAlg(d, m);
    return (s % m + m) % m;  // Handling negative modulo result
}

int main() {
    int m;  
    int num;
    printf("Enter d,mod:\n");
    scanf("%d%d",&num,&m); 
    int s = findMultIn(num, m);

    cout << num << "^-1" << " mod " << m << " = " << s << endl;
    cout << num << " * " << s << " mod " << m << " = 1" << endl;

    return 0;
}

File: C#/ConsolePr2_3.cs
using System;

class Program
{
    static (int, int, int) GenerEuclidAlg(int a, int b)
    {
        if (b == 0)
        {
            return (a, 1, 0);
        }
        else
        {
            int d, x, y;
            (d, x, y) = GenerEuclidAlg(b, a % b);
            return (d, y, x - (a / b) * y);
        }
    }

    static int FindMultIn(int d, int m)
    {
        int s, _;
        (_, s, _) = GenerEuclidAlg(d, m);
        return (s % m + m) % m; // Handling negative modulo result
    }

    static void Main()
    {
        int m, num;
        Console.WriteLine("Enter d,mod:");
        num = int.Parse(Console.ReadLine());
        m = int.Parse(Console.ReadLine());

        int s = FindMultIn(num, m);
        Console.WriteLine($"{num}^-1 mod {m} = {s}");
        Console.WriteLine($"{num} * {s} mod {m} = 1");
    }
}